tests/unit/command/test_diff.py

```diff
@@ -220,6 +220,22 @@ def test_no_changes(mocker, capsys):
     assert not out
 
 
+def test_no_changes_hide_missing(mocker, capsys):
+    args = parse_args(["diff", "--json", "--hide-missing"])
+    cmd = args.func(args)
+    mocker.patch("dvc.repo.Repo.diff", return_value={})
```

Comment:
Please use `pytest.mark.parametrize` in [test_no_changes](https://github.com/dberenbaum/dvc/blob/00d854210f8ff6446335616097a8645b826e04ae/tests/unit/command/test_diff.py#L207).

Reply:
You can change the above test `test_no_changes` to something like following:
```python
@pytest.mark.parametrize("opts", ([], ["a_rev", "b_rev"], ["--hide-missing"]))
def test_no_changes(mocker, capsys, opts):
    args = parse_args(["diff", "--json", *opts])
    cmd = args.func(args)
    mocker.patch("dvc.repo.Repo.diff", return_value={})

    assert 0 == cmd.run()
    out, _ = capsys.readouterr()
    assert "{}" in out

    args = parse_args(["diff", *opts])
    cmd = args.func(args)
    assert 0 == cmd.run()
    out, _ = capsys.readouterr()
    assert not out
```

Reply:
Sorry, just got a little lazy/sloppy here trying to fix this as quickly as possible.

Reply:
@skshetry Updated the tests with parametrized options and output formats.

Reply:
@skshetry Pinging on this. Could you review the updated tests when you have a chance? 🙏